uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/test/java/org/guvnor/m2repo/backend/server/M2ServletContextListenerTest.java

```diff
@@ -0,0 +1,127 @@
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.connector.basic.BasicRepositoryConnectorFactory;
+import org.eclipse.aether.impl.DefaultServiceLocator;
+import org.eclipse.aether.repository.LocalRepository;
+import org.eclipse.aether.resolution.ArtifactRequest;
+import org.eclipse.aether.resolution.ArtifactResolutionException;
+import org.eclipse.aether.resolution.ArtifactResult;
+import org.eclipse.aether.spi.connector.RepositoryConnectorFactory;
+import org.eclipse.aether.spi.connector.transport.TransporterFactory;
+import org.eclipse.aether.transport.file.FileTransporterFactory;
+import org.eclipse.aether.transport.http.HttpTransporterFactory;
+import org.guvnor.common.services.project.model.GAV;
+import org.guvnor.m2repo.backend.server.repositories.ArtifactRepositoryService;
+import org.guvnor.m2repo.preferences.ArtifactRepositoryPreference;
+import org.junit.After;
+import org.junit.Test;
+
+import static org.assertj.core.api.Assertions.*;
+
+public class M2ServletContextListenerTest {
+
+    private static RepositorySystemSession newSession(RepositorySystem system) {
+        ArtifactRepositoryPreference artifactRepositoryPreference = new ArtifactRepositoryPreference();
+        artifactRepositoryPreference.defaultValue(artifactRepositoryPreference);
+        DefaultRepositorySystemSession session = MavenRepositorySystemUtils.newSession();
+        LocalRepository localRepo = new LocalRepository(artifactRepositoryPreference.getGlobalM2RepoDir());
+        session.setLocalRepositoryManager(system.newLocalRepositoryManager(session,
+                                                                           localRepo));
+
+        return session;
+    }
+
+    private static RepositorySystem newRepositorySystem() {
+        DefaultServiceLocator locator = MavenRepositorySystemUtils.newServiceLocator();
+        locator.addService(RepositoryConnectorFactory.class,
+                           BasicRepositoryConnectorFactory.class);
+        locator.addService(TransporterFactory.class,
+                           FileTransporterFactory.class);
+        locator.addService(TransporterFactory.class,
+                           HttpTransporterFactory.class);
+        return locator.getService(RepositorySystem.class);
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        deleteArtifactIFPresent();
+    }
+
+    private void deleteArtifactIFPresent() throws ArtifactResolutionException {
+        ArtifactRequest artifactRequest = new ArtifactRequest();
+        artifactRequest.setArtifact(getArtifact());
+        ArtifactResult result = Aether.getAether().getSystem().resolveArtifact(newSession(newRepositorySystem()),
+                                                                               artifactRequest);
+        if (!result.isMissing()) {
+            File artifactFile = result.getArtifact().getFile();
+            assertThat(artifactFile.delete());
```

Comment:
This asserThat will do nothing - you need to add something like `isTrue()`

Reply:
Fixed